nautobot_netbox_importer/management/commands/import_netbox_objectchange_json.py

```diff
@@ -0,0 +1,134 @@
+            if not self.map_object_type("related_object_type", entry, used_error_messages):
+                return
+        else:
+            del entry["fields"]["related_object_type"]
+
+        modelname = entry["fields"]["changed_object_type"].model
+        entry["fields"]["changed_object_id"] = netbox_pk_to_nautobot_pk(modelname, entry["fields"]["changed_object_id"])
+        if entry["fields"]["related_object_id"]:
+            modelname = entry["fields"]["related_object_type"].model
+            entry["fields"]["related_object_id"] = netbox_pk_to_nautobot_pk(
+                modelname, entry["fields"]["related_object_id"]
+            )
+
+        entry["fields"]["user"] = User.objects.filter(username=entry["fields"]["user_name"]).first()
+        if not entry["fields"]["user"]:
+            error_message = f'Username {entry["fields"]["user_name"]} not present in DB.'
+            if error_message not in used_error_messages:
+                used_error_messages.append(error_message)
+                self.logger.error(error_message)
+            return
+
+        if not options["dry_run"]:
+            obj = ObjectChange.objects.create(**entry["fields"])
+            obj.time = entry["fields"]["time"]
+            obj.full_clean()
+            obj.save()
+
+    def handle(self, *args, **options):
+        """Handle execution of the import_netbox_json management command."""
+        validate_netbox_version(options["netbox_version"])
+
+        self.logger, _ = initialize_logger(options)
+
+        self.logger.info("Loading NetBox JSON data into memory...", filename={options["json_file"].name})
+        no_objectchange_data = json.load(options["json_file"])
+
+        if not isinstance(no_objectchange_data, list):
+            raise CommandError(f"Data should be a list of records, but instead is {type(no_objectchange_data)}!")
+        self.logger.info("JSON data loaded into memory successfully.")
+
+        self.logger.info("Creating NetBox ContentType mapping...")
+        for entry in no_objectchange_data:
+            if entry["model"] == "contenttypes.contenttype":
+                self.netbox_contenttype_mapping[entry["pk"]] = (
+                    entry["fields"]["app_label"],
+                    entry["fields"]["model"],
+                )
+        # Freeing some memory one the object is processed
+        no_objectchange_data = None
+
+        self.logger.info("Creating Nautobot ContentType mapping...")
+        for entry in ContentType.objects.all():
+            self.nautobot_contenttype_mapping[(entry.app_label, entry.model)] = entry.id
+
+        self.logger.info(
+            "Loading ObjectChange NetBox JSON data into memory...", filename={options["objectchange_json_file"].name}
+        )
+        objectchange_data = json.load(options["objectchange_json_file"])
+        # used_error_messages is used to avoid repeating the same error message for each entry
+        used_error_messages = []
```

Comment:
Since you're often checking for membership in this, using `set()` (and `used_error_messages.add()`) instead of `[]` and `used_error_messages.append()` would be more efficient.